src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -133,39 +133,48 @@ namespace OpenMS
   {
   }
 
-
   void TheoreticalSpectrumGenerator::getSpectrum(PeakSpectrum& spectrum, const AASequence& peptide, Int min_charge, Int max_charge) const
   {
     if (peptide.empty())
     {
       return;
     }
 
-    PeakSpectrum::StringDataArray ion_names;
-    PeakSpectrum::IntegerDataArray charges;
+    MSSpectrum::Chunks chunks(spectrum);
+    PeakSpectrum::StringDataArray* ion_names;
+    PeakSpectrum::IntegerDataArray* charges;
 
-    if (add_metainfo_)
+    bool charges_dynamic = false, ion_names_dynamic = false;
```

Comment:
no need to use pointers.
Just initialize 
```
PeakSpectrum::StringDataArray* ion_names = 0;
PeakSpectrum::IntegerDataArray* charges = 0;
```

and call delete on it unconditionally.
(that is safe)